nessie/src/main/java/org/apache/iceberg/nessie/NessieIcebergClient.java

```diff
@@ -181,133 +185,223 @@ public IcebergTable table(TableIdentifier tableIdentifier) {
-      withReference(
-              getApi()
-                  .createNamespace()
-                  .namespace(org.projectnessie.model.Namespace.of(namespace.levels()))
-                  .properties(metadata))
-          .create();
-      refresh();
-    } catch (NessieNamespaceAlreadyExistsException e) {
-      throw new AlreadyExistsException(e, "Namespace already exists: %s", namespace);
+
+      Map<ContentKey, Content> contentMap =
+          api.getContent().reference(getReference()).key(key).get();
+      Content existing = contentMap.get(key);
+      if (existing != null) {
+        throw namespaceAlreadyExists(key, existing, null);
+      }
+
+      try {
+
+        commitRetry("create namespace " + key, Operation.Put.of(key, content));
+
+      } catch (NessieReferenceConflictException e) {
+
+        NessieConflictHandler.handleSingle(
+            e,
+            (conflictType, contentKey) -> {
+              switch (conflictType) {
+                case KEY_EXISTS:
+                  Content conflicting =
+                      withReference(api.getContent()).key(contentKey).get().get(contentKey);
+                  throw namespaceAlreadyExists(contentKey, conflicting, e);
+                case NAMESPACE_ABSENT:
+                  throw new NoSuchNamespaceException(
+                      e,
+                      "Cannot create Namespace '%s': parent namespace '%s' does not exist",
+                      namespace,
+                      contentKey);
+                default:
+                  return false;
+              }
+            });
+        throw new RuntimeException(
+            String.format("Cannot create Namespace '%s': %s", namespace, e.getMessage()));
+      }
+
     } catch (NessieNotFoundException e) {
       throw new RuntimeException(
           String.format(
-              "Cannot create Namespace '%s': " + "ref '%s' is no longer valid.",
+              "Cannot create Namespace '%s': ref '%s' is no longer valid.",
               namespace, getRef().getName()),
           e);
+    } catch (BaseNessieClientServerException e) {
+      throw new RuntimeException(
+          String.format("Cannot create Namespace '%s': %s", namespace, e.getMessage()), e);
     }
   }
 
   public List<Namespace> listNamespaces(Namespace namespace) throws NoSuchNamespaceException {
+
     try {
-      GetNamespacesResponse response =
-          withReference(
-                  getApi()
-                      .getMultipleNamespaces()
-                      .namespace(org.projectnessie.model.Namespace.of(namespace.levels())))
-              .get();
-      return response.getNamespaces().stream()
-          .map(ns -> Namespace.of(ns.getElements().toArray(new String[0])))
-          .filter(ns -> ns.length() == namespace.length() + 1)
+
+      org.projectnessie.model.Namespace root =
+          org.projectnessie.model.Namespace.of(namespace.levels());
+
+      String filter =
+          namespace.isEmpty()
```

Comment:
I'd note that it's not possible to create empty namespaces, so in theory it's not possible to get here with an empty namespace.